pcsx2/GS/Renderers/Metal/tfx.metal

```diff
@@ -1021,21 +1022,37 @@ struct PSMain
 			uint4 denorm_c = uint4(C);
 			uint2 denorm_TA = uint2(cb.ta * 255.5f);
 
-			if (PS_READ16_SRC)
+			if (PS_SHUFFLE_SAME)
 			{
-				C.rb = (denorm_c.r >> 3) | (((denorm_c.g >> 3) & 0x7u) << 5);
-				if (denorm_c.a & 0x80)
-					C.ga = (denorm_c.g >> 6) | ((denorm_c.b >> 3) << 2) | (denorm_TA.y & 0x80);
+				if (PS_READ_BA)
+				{
+					C.ga = (denorm_c.b & 0x7Fu) | (denorm_c.a & 0x80);
+					C.rb = C.ga;
+				}
 				else
-					C.ga = (denorm_c.g >> 6) | ((denorm_c.b >> 3) << 2) | (denorm_TA.x & 0x80);
+				{
+					C.ga = C.rg;
+					C.rb = C.ga;
```

Comment:
I might have buggered this up, it might meant to be rg to ba (which will make more sense), but I don't think I have a test case.